usr.bin/paste/paste.c

```diff
@@ -131,6 +133,16 @@ main(int argc, char *argv[])
 		STAILQ_INSERT_TAIL(&lh, lp, entries);
 	}
 
+	if (cap_enter() < 0 && errno != ENOSYS)
+		err(EXIT_FAILURE, "failed to enter capability mode");
+
+	cap_rights_init(&rights, CAP_READ);
+	STAILQ_FOREACH(lp, &lh, entries) {
+		if (lp->fp && cap_rights_limit(fileno(lp->fp), &rights) < 0 &&
```

Comment:
I would suggest calling cap_rights_limit() right after each fopen() call, rather than having a separate loop. It's easier to reason about that way (open+rights limiting happens in one go), and it's simpler.

Also, I don't think you need the `lp->fp` null check.

Also, I believe you can use caph_rights_limit() and avoid the ENOSYS check.

Reply:
> I would suggest calling cap_rights_limit() right after each fopen() call, rather than having a separate loop. It's easier to reason about that way (open+rights limiting happens in one go), and it's simpler.

You're absolutely right, it's totally valid. Done.

> Also, I don't think you need the lp->fp null check.

True.

> Also, I believe you can use caph_rights_limit() and avoid the ENOSYS check.

True, but since it's concise enough for this small use case, I kept it as is. Let me know if you think it needs to be changed.